### DISCUSSION TITLE
"pathLength" attribute for svg icons

### DISCUSSION THREAD

**Discussion by ALPIXELMaiki** [FortAwesome/Font-Awesome#20927] [Feature Requests]
Hi,

According to this [CSS-Tricks article](https://css-tricks.com/a-trick-that-makes-drawing-svg-lines-way-easier/), and in order to animate the svg path easily, please can you add the pathLength attribute to the 'path' elements during the svg generations ?

`<path fill="..." d="..." **pathLength="1"**></path>`

You have to edit fontawesome.js lines 1602, 1610, 1620, and add this attribute for path elements.
```
if (Array.isArray(vectorData)) {
      element = {
        tag: 'g',
        attributes: {
          class: "".concat(config.familyPrefix, "-").concat(DUOTONE_CLASSES.GROUP)
        },
        children: [{
          tag: 'path',
          attributes: {
            class: "".concat(config.familyPrefix, "-").concat(DUOTONE_CLASSES.SECONDARY),
            fill: 'currentColor',
            d: vectorData[0],
            pathLength:"1"
          }
        }, {
          tag: 'path',
          attributes: {
            class: "".concat(config.familyPrefix, "-").concat(DUOTONE_CLASSES.PRIMARY),
            fill: 'currentColor',
            d: vectorData[1],
            pathLength:"1"
          }
        }]
      };
    } else {
      element = {
        tag: 'path',
        attributes: {
          fill: 'currentColor',
          d: vectorData,
          pathLength:"1"
        }
      };
    }
```

Thank you

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community.

Did you test this feature with a Font Awesome icon?

You can find the source svg here: https://github.com/FortAwesome/Font-Awesome/blob/6.x/svgs/solid/house.svg?short_path=4eaa018

**Comment by ALPIXELMaiki** [comment]
Yes, i tested this feature, and it works lie a charm !
"Drawing" icons is fun and really nice to see :)

**Comment by ALPIXELMaiki** [comment]
https://user-images.githubusercontent.com/1275698/155275077-3e67a3bf-1712-4ec5-ab31-d04b992871f3.mov

**Comment by ALPIXELMaiki** [comment]
I just created a codePen for the demo :
https://codepen.io/alpixelmaiki/pen/WNXyKBw
